src/ray/raylet/scheduling/cluster_task_manager.cc

```diff
@@ -277,16 +287,69 @@ bool ClusterTaskManager::CancelTask(const TaskID &task_id) {
 
 void ClusterTaskManager::Heartbeat(bool light_heartbeat_enabled,
                                    std::shared_ptr<HeartbeatTableData> data) const {
+  if (max_resource_shapes_per_load_report_ == 0) {
+    return;
+  }
   // TODO (WangTao): Find a way to check if load changed and combine it with light
   // heartbeat. Now we just report it every time.
   data->set_resource_load_changed(true);
   auto resource_loads = data->mutable_resource_load();
   auto resource_load_by_shape =
       data->mutable_resource_load_by_shape()->mutable_resource_demands();
 
-  // TODO (Alex): Implement the 1-CPU task optimization.
+  int num_reported = 0;
+
+  // 1-CPU optimization
```

Comment:
Comment why we handle it separately.

Reply:
@stephanie-wang what was the reason for this? IIRC it had something to do with 1-cpu tasks being the most common, but in hindsight, I'm not sure because they're also the most easily fulfilled by any node (therefore less important to report).